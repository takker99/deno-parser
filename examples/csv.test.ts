import { CSV } from "./csv.ts";
import { assertSnapshot } from "@std/testing/snapshot";

Deno.test("csv simple", (t) =>
  assertSnapshot(
    t,
    CSV.parse(
      `\
apple,1.23,1312
banana,0.99,67
cherry,0.54,987
dragonfruit,2.87,4
elderberry,9.99,22
`,
    ),
  ));

Deno.test("csv complex", (t) =>
  assertSnapshot(
    t,
    CSV.parse(
      `\
a,,c,"a ""complex"" field, i think"
d,eeeeee,FFFF,cool
nice,nice,nice3,nice4
`,
    ),
  ));

Deno.test("csv LF", (t) =>
  assertSnapshot(
    t,
    CSV.parse(
      `\
a,b,c\n\
a,b,c\n\
a,b,c\
`,
    ),
  ));

Deno.test("csv CRLF", (t) =>
  assertSnapshot(
    t,
    CSV.parse(
      `\
a,b,c\r\n\
a,b,c\r\n\
a,b,c\
`,
    ),
  ));

Deno.test("csv trailing newline", async (t) => {
  await assertSnapshot(
    t,
    CSV.parse(
      `\
a,b,c\r\n\
a,b,c\r\n\
a,b,c\r\n\
`,
    ),
  );
  await assertSnapshot(
    t,
    CSV.parse(
      `\
a,b,c\n\
a,b,c\n\
a,b,c\n\
`,
    ),
  );
});
