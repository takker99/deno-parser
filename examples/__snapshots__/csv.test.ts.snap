export const snapshot = {};

snapshot[`csv simple 1`] = `
{
  type: "ParseOK",
  value: [
    [
      "apple",
      "1.23",
      "1312",
    ],
    [
      "banana",
      "0.99",
      "67",
    ],
    [
      "cherry",
      "0.54",
      "987",
    ],
    [
      "dragonfruit",
      "2.87",
      "4",
    ],
    [
      "elderberry",
      "9.99",
      "22",
    ],
  ],
}
`;

snapshot[`csv complex 1`] = `
{
  type: "ParseOK",
  value: [
    [
      "a",
      "",
      "c",
      'a "complex" field, i think',
    ],
    [
      "d",
      "eeeeee",
      "FFFF",
      "cool",
    ],
    [
      "nice",
      "nice",
      "nice3",
      "nice4",
    ],
  ],
}
`;

snapshot[`csv LF 1`] = `
{
  type: "ParseOK",
  value: [
    [
      "a",
      "b",
      "c",
    ],
    [
      "a",
      "b",
      "c",
    ],
    [
      "a",
      "b",
      "c",
    ],
  ],
}
`;

snapshot[`csv CRLF 1`] = `
{
  type: "ParseOK",
  value: [
    [
      "a",
      "b",
      "c",
    ],
    [
      "a",
      "b",
      "c",
    ],
    [
      "a",
      "b",
      "c",
    ],
  ],
}
`;

snapshot[`csv trailing newline 1`] = `
{
  type: "ParseOK",
  value: [
    [
      "a",
      "b",
      "c",
    ],
    [
      "a",
      "b",
      "c",
    ],
    [
      "a",
      "b",
      "c",
    ],
  ],
}
`;

snapshot[`csv trailing newline 2`] = `
{
  type: "ParseOK",
  value: [
    [
      "a",
      "b",
      "c",
    ],
    [
      "a",
      "b",
      "c",
    ],
    [
      "a",
      "b",
      "c",
    ],
  ],
}
`;
